File: system.py
import os
import time
from google.oauth2 import service_account
from googleapiclient.discovery import build
import vlc  # For audio playback

# Google Drive API credentials (Service Account)
SERVICE_ACCOUNT_FILE = 'path_to_service_account_credentials.json'  # Replace with your service account credentials file
SCOPES = ['https://www.googleapis.com/auth/drive']

# ID of the folder containing song files in Google Drive
FOLDER_ID = 'your_folder_id_here'  # Replace with your folder ID

# Function to authenticate and retrieve song files from Google Drive
def fetch_song_files():
    try:
        creds = service_account.Credentials.from_service_account_file(SERVICE_ACCOUNT_FILE, scopes=SCOPES)
        service = build('drive', 'v3', credentials=creds)

        song_files = []
        page_token = None

        while True:
            response = service.files().list(q=f"'{FOLDER_ID}' in parents",
                                            fields='nextPageToken, files(name, webViewLink)').execute()
            files = response.get('files', [])
            for file in files:
                song_files.append({
                    'name': file['name'],
                    'webLink': file['webViewLink']
                })

            page_token = response.get('nextPageToken', None)
            if page_token is None:
                break

        return song_files
    except Exception as e:
        print(f"An error occurred while fetching song files: {e}")
        return []

# Function to download a song file from Google Drive
def download_song_file(song):
    try:
        os.system(f"wget -O '{song['name']}' '{song['webLink']}'")
    except Exception as e:
        print(f"An error occurred while downloading the song '{song['name']}': {e}")

# Function to play songs using VLC media player
def play_song(song):
    try:
        instance = vlc.Instance('--input-repeat=-1', '--fullscreen')  # VLC instance with repeat and fullscreen
        player = instance.media_player_new()
        media = instance.media_new(song)
        media.get_mrl()
        player.set_media(media)
        player.play()
        time.sleep(5)  # Play each song for 5 seconds (adjust as needed)
        player.stop()
    except Exception as e:
        print(f"An error occurred while playing the song '{song}': {e}")

# Fetch song files from Google Drive and download them
song_files = fetch_song_files()
for song in song_files:
    download_song_file(song)

# Function to display the music player dashboard menu
def display_dashboard():
    print("===============================")
    print("       Music Player Menu       ")
    print("===============================")
    print("1. Play all songs")
    print("2. Exit")
    print("===============================")

# Function to handle user choice
def handle_choice(choice):
    if choice == '1':
        print("========== Autoplaying Music ==========")
        if song_files:
            print("Welcome to the Music Player!")
            print(f"Found {len(song_files)} songs in your Google Drive folder.")
            print("Enjoy the music!\n")

            for idx, song in enumerate(song_files, start=1):
                print(f"Now playing: {song['name']} ({idx}/{len(song_files)})")
                play_song(song['name'])
                if idx < len(song_files):
                    input("Press Enter to play the next song...")
            
            print("\nAll songs have been played. Hope you enjoyed the music!")
        else:
            print("No songs found in the Google Drive folder.")
    elif choice == '2':
        print("Exiting the Music Player...")
        # Add any necessary cleanup code or exit statements here
        pass
    else:
        print("Invalid choice. Please enter a valid option.")

# Display the dashboard and handle user choice
while True:
    display_dashboard()
    user_choice = input("Enter your choice (1-2): ")
    handle_choice(user_choice)
    if user_choice == '2':
        break

File: system.cpp
#include <iostream>
#include <vector>
#include <string>

// Song structure with title, artist, and description
struct Song {
    std::string title;
    std::string artist;
    std::string description;
};

// Function to display the music player dashboard menu
void displayMenu() {
    std::cout << "===== Music Player Dashboard =====" << std::endl;
    std::cout << "1. Play a song" << std::endl;
    std::cout << "2. Display playlist" << std::endl;
    std::cout << "3. Exit" << std::endl;
    std::cout << "=================================" << std::endl;
    std::cout << "Enter your choice: ";
}

// Function to clear the console screen
void clearScreen() {
    std::cout << "\033[2J\033[1;1H"; // ANSI escape codes to clear screen
}

int main() {
    // Create a vector to store songs
    std::vector<Song> playlist;

    // Add 10 sample songs with titles, artists, and descriptions
    Song song1 = {"Song Title 1", "Artist 1", "Description for Song 1"};
    Song song2 = {"Song Title 2", "Artist 2", "Description for Song 2"};
    Song song3 = {"Song Title 3", "Artist 3", "Description for Song 3"};
    Song song4 = {"Song Title 4", "Artist 4", "Description for Song 4"};
    Song song5 = {"Song Title 5", "Artist 5", "Description for Song 5"};
    Song song6 = {"Song Title 6", "Artist 6", "Description for Song 6"};
    Song song7 = {"Song Title 7", "Artist 7", "Description for Song 7"};
    Song song8 = {"Song Title 8", "Artist 8", "Description for Song 8"};
    Song song9 = {"Song Title 9", "Artist 9", "Description for Song 9"};
    Song song10 = {"Song Title 10", "Artist 10", "Description for Song 10"};

    // Add songs to the playlist
    playlist.push_back(song1);
    playlist.push_back(song2);
    playlist.push_back(song3);
    playlist.push_back(song4);
    playlist.push_back(song5);
    playlist.push_back(song6);
    playlist.push_back(song7);
    playlist.push_back(song8);
    playlist.push_back(song9);
    playlist.push_back(song10);

    int choice;
    do {
        clearScreen(); // Clear the screen
        displayMenu(); // Display dashboard menu
        std::cin >> choice;

        switch(choice) {
            case 1: {
                clearScreen();
                std::cout << "=== Play a Song ===" << std::endl;
                std::cout << "Select a song to play:" << std::endl;
                for (size_t i = 0; i < playlist.size(); ++i) {
                    std::cout << i + 1 << ". " << playlist[i].title << " - " << playlist[i].artist << std::endl;
                }
                std::cout << "Enter song number: ";
                int songNumber;
                std::cin >> songNumber;
                if (songNumber >= 1 && songNumber <= static_cast<int>(playlist.size())) {
                    clearScreen();
                    std::cout << "Now playing: " << playlist[songNumber - 1].title << " - " << playlist[songNumber - 1].artist << std::endl;
                    std::cout << "Description: " << playlist[songNumber - 1].description << std::endl;
                } else {
                    std::cout << "Invalid song number." << std::endl;
                }
                std::cout << "Press Enter to continue...";
                std::cin.ignore();
                std::cin.get();
                break;
            }
            case 2: {
                clearScreen();
                std::cout << "=== Playlist ===" << std::endl;
                for (size_t i = 0; i < playlist.size(); ++i) {
                    std::cout << "Song " << i + 1 << ": " << playlist[i].title << " - " << playlist[i].artist << std::endl;
                    std::cout << "Description: " << playlist[i].description << std::endl;
                    std::cout << std::endl;
                }
                std::cout << "Press Enter to continue...";
                std::cin.ignore();
                std::cin.get();
                break;
            }
            case 3:
                clearScreen();
                std::cout << "Exiting the music player..." << std::endl;
                break;
            default:
                clearScreen();
                std::cout << "Invalid choice. Please enter a valid option." << std::endl;
                std::cout << "Press Enter to continue...";
                std::cin.ignore();
                std::cin.get();
        }
    } while (choice != 3);

    return 0;
}
